### PR TITLE
Remove duplicates from recommendations

### PR DESCRIPTION
**Pull request by anuprulez**
Remove duplicates from recommendations in the workflow editor.

ping @gmauro 

thanks!

### CODE REVIEW

**Comment by anuprulez** [comment]
Thanks @gmauro

File: client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import { getToolPredictions } from "./services";
 import LoadingSpan from "components/LoadingSpan";
 import _l from "utils/localization";
+import _ from "underscore";
```
Comment (bgruening) [inline client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: @anuprulez is there no way to not use underscore? We want to get rid of this library in the long run and just use standard JS.

Also can you make sure you upstream the changes? Thanks.

Comment (anuprulez) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: Ok, the changes will be done in the upstream. Do you know why "underscoreJS" will be removed? @bgruening

Comment (bgruening) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: I think the idea is to totally get rid of underscore and jQuery if possible as most of the functionality is now available in ES6.

Comment (anuprulez) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: Yes, it makes sense. Thanks!

Comment (bgruening) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: @anuprulez btw. why do you filter on the client? Should the API not filter that and __not__ return the duplicates?

Comment (anuprulez) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: The API returns a unique list but the datatype compatibility is checked on the client-side (I use the same logic which is being used to check datatype compatibility while connecting toolboxes on the workflow editor i.e. showing orange or green colour). This part of the code needs to be refactored which erroneously adds duplicate items while checking for compatibility between all input (of all the recommended tools) and output datatypes (of the last tool).

Comment (anuprulez) [reply client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue:28]: And this part of the code is currently separate for tool recommendations shown after a tool execution (here this case was handled) and in the workflow editor (here it was not). Therefore, duplicate ones were shown in the workflow editor but not after tool execution for the same tool. This needs to be unified. I will do all this refactoring and push it to upstream. No changes in the API will be needed in my opinion, just the UI ones

**Pull request merged**